### PR TITLE
Use SDL_RWops for file access where possible

### PR DESCRIPTION
**Pull request by ceckak**
Some platforms have nontraditional filesystem access and SDL has already provided an abstraction layer for them in the form of [SDL_RWops](https://wiki.libsdl.org/SDL_RWops).  Rather than introduce platform-specific code into exult to account for platform filesystem differences, leverage SDL's abstraction as the primary interface for interacting with the filesystem.

SDL_rwops was officially introduced in SDL 2.0.10, so this patch also updated the minimum SDL version to that level, and fixes a minor bug in the SDL version checking in `acinclude.m4`.

Note that there is one glaring deficiency in `SDL_RWops`: it does not provide an api for interacting with directories.  So while this change works for reading and writing individual files, it does not help with searching directories to see what they contain, so we end up falling back to legacy code that bypasses SDL, which might not work on all platforms.  This might cast doubt on whether there is value in switching to `SDL_RWops`, but the only place exult currently queries directories is when detecting mods.  This means mods need to live in a traditional filesystem location, but all other content can be placed in platform-specific locations.  This works out well for android where some of the data files can be bundled directly into the apk, while mods are installed into the device's filesystem where they can be detected.

### CODE REVIEW

**Comment by ceckak** [comment]
Fixed the link error on windows (it wasn't pulling in the  new `sdlrwops*` sources).  Presumably the same issue exists for ios, but I'm not set up to edit the ios project to add them.  Might need some help there if we want to include that with this commit, otherwise we can circle back and fix the ios build separately after the dust settles from the android merge.

**Comment by DominusExult** [comment]
I can quickly fix it for the iOS port once this is merged.

Can anyone please review this series of pull requests? I'm really not equipped to do this (other than reporting that a qick test seemed fine). @marzojr @drcode1 @Dragon-Baroque

**Comment by Dragon-Baroque** [comment]
Sure willing to help you, @DominusExult.
Now what is the most efficient way to bring a pull request into my local Git trees from a fork [ `ceckak/exult.git` ] that my Git installed trees don't know about [ I have locally installed reference `Exult/exult.git` and my own fork `Dragon-Baroque/exult.git` ] ?

**Comment by DominusExult** [comment]
Thanks! Last time I just checked out the fork into a new folder, tried there and then merged from the github page.
But I think you could also add this fork as a remote. But as ceckak changes his fork "destructively" at times the first way might be better to keep your working copy clean and unbroken ;)

**Comment by ceckak** [comment]
> as ceckak changes his fork "destructively" at times

Guilty as charged.  I've been managing this for as a way to distribute a patch series that I'm continuing to refine until they get merged, but I can pivot to preserving history if that makes things easier for you guys.

**Comment by DominusExult** [comment]
I'm fine with it ;)
I just noticed it when I suddenly had an abandoned fork with my iOS fixes for exactly this issue a while ago ;)

**Comment by Dragon-Baroque** [comment]
> but I can pivot to preserving history if that makes things easier for you guys.

Thanks for the offer, but not needed. I did exactly the same in my own Pull Requests. Only I am now on the receiving side, and @DominusExult essentially stated that I should take caution if I used the `remote` strategy as compared to the `clone` strategy.

**Comment by ceckak** [comment]
> also his PRs always contains stuff from other PRs. for example this one contains the sdl_main commit that was filed as a separate PR. if this one here is merged as-is, there will be 2 commits "Remove sdl_main.cc wrapper"
which is not only confusing to the reader of the git history, but it will also break automated tools assuming a flat/continuous history.

I would love to fix that if someone can tell me how.  I have a series of 46 patches which largely depend on each other and need to be merged in series.  I didn't want to submit a single PR with all of them since it makes it hard for you to review them independently, but it also seemed helpful to submit several related PRs at a time.  I have yet to find a way to do that which doesn't show the duplication you're referring to.  I am happy to rebase the later PRs as the earlier ones get merged though, which eliminates the duplication while retaining the comment thread on the PR.

**Comment by rofl0r** [comment]
> I would love to fix that if someone can tell me how.

there's only two ways to achieve that i'm aware of:

- a single PR containing all patches
- a single PR at a time, wait for merge, file next

**Comment by ceckak** [comment]
Ok - that's consistent with what I've been seeing.  Feel free to ignore the later PRs if they aren't useful for context; I will rebase them as we merge from the head of the queue.

**Comment by Dragon-Baroque** [comment]
Have you noticed that the compilation checks on Linux failed ?
This is not your fault, but you could fix it :
- You require SDL 2.0.10 in `acinclude.m4` [ was 2.0.6 before the commit ],
- The Ubuntu Linux installs 2.0.10 _exactly,_
- The version test line 72 in `acinclude.m4` is wrong, it uses a `-gt` for the Patchlevel, should use a `-ge.`

Could you fix this and retry ? That should pass the Linux compilation tests.

**Comment by ceckak** [comment]
> Have you noticed that the compilation checks on Linux failed ? This is not your fault, but you could fix it :
> 
> * You require SDL 2.0.10 in `acinclude.m4` [ was 2.0.6 before the commit ],
> * The Ubuntu Linux installs 2.0.10 _exactly,_
> * The version test line 72 in `acinclude.m4` is wrong, it uses a `-gt` for the Patchlevel, should use a `-ge.`
> 
> Could you fix this and retry ? That should pass the Linux compilation tests.

Thanks for chasing that down.  I had noticed the build failures, but hadn't realized the SDL version change was triggering it.  I'll add a fix for the version test as you've suggested.

File: files/sdlrwopsstreambuf.cc
Comment (marzojr) [inline files/sdlrwopsstreambuf.cc:223]: We already know width is 1 here, so the check `width <= 0` is always false (see [LGTM 1](https://lgtm.com/projects/g/exult/exult/snapshot/a28cf7660fd554c904f32373c65983cdbc5fb585/files/files/sdlrwopsstreambuf.cc#xf56c03303f434649:1).

File: files/sdlrwopsstreambuf.cc
Comment (marzojr) [inline files/sdlrwopsstreambuf.cc:239]: We already know width is 1 here, so the ternary operator can be replaced by `width * off` (see [LGTM 2](https://lgtm.com/projects/g/exult/exult/snapshot/a28cf7660fd554c904f32373c65983cdbc5fb585/files/files/sdlrwopsstreambuf.cc#x6edf1c4e1249dcaa:1).

**Review by marzojr — CHANGES_REQUESTED** [review]

**Review by marzojr — APPROVED** [review]

**Pull request merged**
